wp-content/civi-extensions/goonjcustom/Civi/CollectionCampHelper.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Civi;
+
+use Civi\Core\Service\AutoSubscriber;
+use CRM_Pets_ExtensionUtil as E;
+
+class CollectionCampHelper extends AutoSubscriber {
+
+	public static function getSubscribedEvents() {
+		return [
+			'&hook_civicrm_post' => 'generateCollectionCampCode',
+		];
+	}
+
+	/**
+	 * This hook is called after a db write on entities.
+	 *
+	 * @param string $op
+	 *   The type of operation being performed.
+	 * @param string $objectName
+	 *   The name of the object.
+	 * @param int $objectId
+	 *   The unique identifier for the object.
+	 * @param object $objectRef
+	 *   The reference to the object.
+	 */
+	public static function generateCollectionCampCode(string $op, string $objectName, int $objectId, &$objectRef) {
+		// Check if the object name is 'AfformSubmission'
+		if ($objectName !== 'AfformSubmission') {
+			return;
+		}
+
+		// Extract the 'data' field
+		$data = $objectRef->data;
+		$decodedData = json_decode($data, true);
+
+		// Access the subtype
+		$subtype = $decodedData['Eck_Collection_Camp1'][0]['fields']['subtype'] ?? null;
+
+		// Check if the subtype is 4 or 5
+		if ($subtype != 4 && $subtype != 5) {
+			return;
+		}
+
+		// Access the id within the decoded data
+		if (!isset($decodedData['Eck_Collection_Camp1'][0]['fields']['id'])) {
```

Comment:
Why have we hardcoded the 0th index?

Reply:
When we are submitting the form where I'm getting the data in the below format so I can simply catch the 0th index and find the corresponding item from there, that's why I'm using the `['Eck_Collection_Camp1'][0]['fields']['id']))`.

I have changed it to something like where first I catch the fields at the top of the code `['Eck_Collection_Camp1'][0]['fields']` then use the variable at other places to find the items from that.

```
[26-Aug-2024 12:44:32 Asia/Kolkata] decodedData: Array
(
    [extra] => Array
        (
        )

    [Eck_Collection_Camp1] => Array
        (
            [0] => Array
                (
                    [fields] => Array
                        (
                            [id] => 48
                            [Collection_Camp_Core_Details.Contact_Id] => 8256
                            [Dropping_Centre.Location] => fsda
                            [Dropping_Centre.State] => 1098
                            [Collection_Camp_Core_Details.Status] => 1
                            [title] => Dropping Center
                            [subtype] => 5
                            [Collection_Camp_Core_Details.Authorizer] => 
                            [Collection_Camp_Core_Details.Authorization_date] => now
                        )

                    [joins] => Array
                        (
                        )

                )

        )

)
```